python/ql/src/semmle/python/Pruning.qll

```diff
@@ -182,11 +182,8 @@ module Pruner {
          */
         abstract predicate constrainsVariableToBe(boolean value);
 
-        /** Holds if this constraint implies that `other` cannot hold for the
-         * same variable.
-         * For example `x > 0` implies that `not bool(x)` is `False`.
-         */
-        abstract predicate contradicts(Constraint other);
+        /** Holds if this constraint implies that `other` cannot be `None` */
```

Comment:
This should probably be something like ``if the value constrained by this constraint cannot be `None` ``.

Reply:
done